### PR TITLE
Update conversation SDK to v4.0.2 

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
Updates conversation SDK to v4.0.2

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by rusmonster** [comment]
In scope of migration to v4.0.2 we also should drop `TwilioExtensions.kt`, instead we should use extensions from the SDK taking into account that they throw `TwilioException`, not `ConversationException`

**Comment by ShivamPokhriyal** [comment]
> In scope of migration to v4.0.2 we also should drop `TwilioExtensions.kt`, instead we should use extensions from the SDK taking into account that they throw `TwilioException`, not `ConversationException`

Just a note - There are a few extensions which are not yet added in the SDK.

File: app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt [hunk 1]
```diff
@@ -176,305 +84,49 @@ suspend fun Conversation.getUnreadMessageCount(): Long? = suspendCancellableCoro
...
-}
-
-suspend fun Conversation.unmuteConversation(): Unit = suspendCoroutine { continuation ->
-    setNotificationLevel(Conversation.NotificationLevel.DEFAULT, object : StatusListener {
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
+suspend fun Conversation.waitForSynchronization(): Conversation {
```
Comment (ShivamPokhriyal) [inline app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt:91]: `waitForSynchronization` is actually available from SDK, but it doesn't return anything there. 
@rusmonster Should I update the SDK method to return `Conversation` instance?

Comment (rusmonster) [reply app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt:100]: No, it's still usable in chains like `apply { waitForSynchronization() }`

**Comment by rusmonster** [comment]
lgfm, but wait until we recover SI. Just to make sure tests are green

File: app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt (hunk 1 above)
Comment (ShivamPokhriyal) [reply app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt:100]: Ahh, my bad, totally forgot about kotlin utilities.

Comment (ShivamPokhriyal) [reply app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt]: Applied the suggestion

**Comment by rusmonster** [comment]
pipeline is recovered, but fix this plz before merge:
```
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 50): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 61): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 73): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 84): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 94): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (129, 105): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (130, 83): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (130, 94): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (130, 101): Elvis operator (?:) always returns the left operand of non-nullable type Int
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (141, 83): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (141, 94): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (141, 101): Elvis operator (?:) always returns the left operand of non-nullable type Int
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 55): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 66): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 78): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 89): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 99): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (154, 110): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (155, 87): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (155, 98): Unnecessary safe call on a non-null receiver of type ErrorInfo
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (155, 105): Elvis operator (?:) always returns the left operand of non-nullable type Int
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (166, 83): Safe call on a non-null receiver will have nullable type in future releases
w: /root/convo-demo-app/app/src/main/java/com/twilio/conversations/app/viewModel/MessageListViewModel.kt: (166, 94): Unnecessary safe call on a non-null receiver of type ErrorInfo

```

**Review by rusmonster — CHANGES_REQUESTED** [review]
To fix: `Unnecessary safe call` warnings

**Comment by rusmonster** [comment]
also update `This demo app SDK version:` in the README

**Comment by ShivamPokhriyal** [comment]
@rusmonster Made the changes, please review again 🥺

**Review by rusmonster — APPROVED** [review]

**Pull request merged**